core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -85,19 +86,23 @@ public Schema schema() {
 
     private AllDataFilesTableScan(
         TableOperations ops, Table table, Long snapshotId, Schema schema, Expression rowFilter,
-        boolean caseSensitive, boolean colStats, Collection<String> selectedColumns, Schema fileSchema,
+        boolean ignoreResiduals, boolean caseSensitive, boolean colStats,
+        Collection<String> selectedColumns, Schema fileSchema,
         ImmutableMap<String, String> options) {
-      super(ops, table, snapshotId, schema, rowFilter, caseSensitive, colStats, selectedColumns, options);
+      super(
+          ops, table, snapshotId, schema, rowFilter, ignoreResiduals,
+          caseSensitive, colStats, selectedColumns, options);
```

Comment:
What about not supporting this option for metadata tables? It doesn't seem necessary and if we can simplify this PR, that would be helpful.

Reply:
I don't have a strong opinion here. We have to check whether we can hit this filtering problem during our actions that rewrite metadata in the future. In metadata tables, we have either static tasks or we read Avro right now. So, we cannot really skip parts of a file like in Parquet, but it is still important to think this through. What if something changes and we start filtering out records on our side before passing records to query engines or write some metadata as Parquet? Static metadata tables have `alwaysTrue` as their data filter. Others, on the other hand, do have a valid data filter passed further that can cause issues.

Reply:
I think it's fine if this is already done. You make a good point about possibly needing it later.